Attribution reporting: two new blogposts (#1736)
GoogleChrome/developer.chrome.com: site/en/docs/privacy-sandbox/attribution-reporting-event-introduction/index.md

## Before you read this

If you're new to Attribution Reporting, head over to [Attribution Reporting in five minutes](/docs/privacy-sandbox/attribution-reporting-in-short/) for a quick overview.
For an in-depth introduction, check out [Introduction to Attribution Reporting](/docs/privacy-sandbox/attribution-reporting-introduction/).

## Use cases for event-level reports

The Attribution Reporting API enables sites to measure conversions in the following cases:

- Ad clicks.
- Ads in a third-party iframe, such as ads on a publisher site that uses a third-party adtech provider.
- Ads in a first-party context, such as ads on a social network or a search engine results page, or a publisher serving their own ads.

This API can generate two types of insights: [aggregate reports](/docs/privacy-sandbox/attribution-reporting-introduction/#:~:text=or%20invalid%20activity.-,Aggregate%20reports,-%2C%20on%20the%20other), and event-level reports. This post focuses on the latter.

**Event-level reports** associate an ad click with coarse conversion data.

<figure>

>   {% Img src="image/O2RNUyVSLubjvENAT3e7JSdqSOx1/WQi0OKAUfD6Agqx9j3s5.png", alt="ALT_TEXT_HERE", width="800", height="570" %}

Author: Added!

  detection and analysis, by allowing you to understand patterns that can be used to
  identify spammy or invalid activity.

## Glossary

- An _attribution_ is the identification of user actions that contribute to an outcome. For example, a purchase can be _attributed_ to an ad click.
- A _conversion_ is a user action that is desirable for the advertiser or adtech company,
  such as a sign-up or purchase.
- In this API, the term _source_ is used to describe what happens on the ad display side⏤typically, a publisher site.
- The term _trigger_ is used to describe what happens on the
  conversion side—typically, an advertiser site.
- An _attribution source event_ (or _attribution source_, or _source_ for short) is when
  a user clicks an ad.
- A _source_ is said to be _triggered_ for attribution when a user converts. An
  _attribution trigger event_ is a conversion.

## How does it work?

To generate event-level reports, the browser matches _attribution source events_ (clicks) with _attribution trigger data_ (conversion data) defined by an adtech. Later, the browser sends the reports to a predefined endpoint, with some delay and noise.

<figure>

> {% Img src="image/O2RNUyVSLubjvENAT3e7JSdqSOx1/LmX5xkFFMmgG8cJEm78I.png", alt="ALT_TEXT_HERE", width="800", height="517" %}

Author: Gotcha thanks, TIL 👍

 <figcaption>How Attribution Reporting with event-level reports works</figcaption>
</figure>
 
{% Details %} 
{% DetailsSummary 'h3' %} 
How it works in detail: event-level reports 
{% endDetailsSummary %} 

> Ad elements (`a`) can be configured with attributes that are specific to attribution reporting:

Author: Fixed 👍

 
- Custom data to attach to an ad click on the publisher's side, for example a
 click ID or campaign ID // `attributionsourceeventid`
- The site for which an attribution trigger (conversion) is expected for this ad. // `attributiondestination`
- The reporting endpoint that will be notified of conversions, namely the endpoint that will receive reports sent by the user's browser. // `attributionreportto`
- Optionally, the duration for which conversions can be attributed to a click on this ad—from the moment the ad is clicked. // `attributionexpiry`

> - Optionally, a priority to assign to this source. // `attributionsourcepriority`

Author: SG! Done

```

{% Aside 'caution' %}
Event-level reports can **not** contain detailed conversion data (trigger data) such as exact purchase values. You **can't** get event-level reports that look like this:

```text
ClickID 1298765678762 led to a $317 purchase (red sports shoes) in Boston, CA.
```

This is supported in [aggregate reports](/docs/privacy-sandbox/attribution-reporting-introduction/#:~:text=Aggregate%20reports%2C%20on%20the%20other%20hand%2C%20offer%20more%20detailed%20conversion%20data).
{% endAside %}

### One reporting endpoint

Reports can be sent only to **one endpoint**.

{% Aside %}
More flexible endpoint configurations are [under discussion](https://github.com/WICG/conversion-measurement-api/issues/96).
{% endAside %}

### Noise

The trigger data (conversion data) is _noised_. Noise is a privacy protection that comes on top of the [data limitation to 3 bits](#detailed-source-side-data-and-limited-conversion-data).

{% Aside "key-term" %}
When trigger data has its values randomized in some way, this creates _noised data_.
{% endAside %}

This means that **a small percentage of the time, a report will contain inaccurate trigger data instead of the real data**.

> In Chrome, noising works as follows: 5% of the time, inaccurate trigger data will be sent instead of the real trigger data.

Reviewer: ```suggestion
In Chrome, 5% of the time the browser will send inaccurate trigger data instead of real.
```

#### Example

ClickID `1298765678762` led to a conversion of value `1`.

Instead of receiving this report:

```text
Click 1298765678762 led to a conversion of value 1.
```

You may receive the following report:

```text
Click 1298765678762 led to a conversion of value 4.
```

### Report limitation

To protect user privacy, the number of reports is limited. For a given source event (click), **at most three conversions can be registered**.

{% Aside  %}
This means that conversions will be dropped if a user converts more than three times. You can choose which conversions to get and which are fine to drop by configuring [conversion priorities](/docs/privacy-sandbox/attribution-reporting-event-guide/#prioritize-specific-conversions).
{% endAside %}

#### Example

A user clicks an ad. Later on, they visit the advertiser site (first conversion), then
create an account (second), and finally make two separate purchases (third and fourth).

Even though they have converted four times, by default, you'll receive a report only for the first three conversions.

### Report scheduling

> Reports are **not sent immediately after a user converts**. They're sent with a delay. This is a privacy protection.

Reviewer: ```suggestion
Reports are **not sent immediately after a user converts**. To protect privacy, reports are sent with a delay during a **reporting window**.
```

> After the initial ad click, a schedule of **reporting windows** begins. Each reporting window has a

Reviewer: ```suggestion
After the initial ad click, a schedule of reporting windows begins. Each reporting window has a
```

deadline. Conversions registered before that deadline will be sent at the end of that window.

{% Aside 'caution' %}
Reports may not be exactly sent at the scheduled dates. If the browser isn't running
when a report is scheduled to be sent, the report is sent at browser startup—which could be days or
weeks after the initially scheduled time.
{% endAside %}

> After expiry (click time + `attributionexpiry`), no additional attribution can be triggered (no additional conversion can be counted).

Reviewer: Trying spelling out more precisely what you mean in these parenthesis. I think it's the duration of time specified in `attributionexpiry`, starting from the click event?

#### Example

> If a user clicks an ad and converts three days later, you may receive the report about seven days

Reviewer: ```suggestion
If a user clicks an ad and converts three days later, you may receive the report about it seven days
```

Author: I meant approximately seven days, fixed.

later.

{% Details %}
{% DetailsSummary 'h3' %}
Report scheduling: detail
{% endDetailsSummary %}

In Chrome, report scheduling for clicks works as follows:

> <div class="w-table-wrapper"><table class="w-table--top-align"><thead><tr><th>Value of <code >attributionexpiry</code><br></th><th>Depending on the conversion time, a report is sent (if the browser is open)...</th><th>Number of reporting windows</th></tr></thead><tbody><tr><td>30 days (the default and maximum value)</td><td><ul><li>2 days after the ad was clicked</li><li>or 7 days after click</li><li>or ((<code>attributionexpiry</code>= 30 days) + 1 hour) after click.</li></ul></td><td>3</td></tr><tr><td>Between 7 and 30 days</td><td><ul><li>2 days after click</li><li>or 7 days after click</li><li>or (<code>attributionexpiry</code> + 1 hour) after click.</li></ul></td><td>3</td></tr><tr><td>Between 2 and 7 days</td><td><ul><li>2 days after click</li><li>or (<code>attributionexpiry</code> + 1 hour) after click.</li></ul></td><td>2</td></tr><tr><td>Under 2 days</td><td><ul><li>(2 days + 1 hour) after click</li></ul></td><td>1</td></tr></tbody></table></div>

Reviewer: ```suggestion
<div class="w-table-wrapper">
    <table class="w-table--top-align">
        <thead>
            <tr>
                <th>Value of <code>attributionexpiry</code><br></th>
                <th>Time a report is sent, depending on the conversion time and whether the browser is open</th>
                <th>Number of reporting windows</th>
            </tr>
        </thead>
        <tbody>
            <tr>
                <td>30 days (the default and maximum value)</td>
                <td>
                    <ul>
                        <li>2 days after the click</li>
                        <li>or 7 days after the click</li>
                        <li>or at maximum, 30 days + 1 hour after the click</li>
                    </ul>
                </td>
                <td>3</td>
            </tr>
            <tr>
                <td>Between 7 and 30 days</td>
                <td>
                    <ul>
                        <li>2 days after the click</li>
                        <li>or 7 days after the click</li>
                        <li>or the value of <code>attributionexpiry</code + 1 hour after the click</li>
                    </ul>
                </td>
                <td>3</td>
            </tr>
            <tr>
                <td>Between 2 and 7 days</td>
                <td>
                    <ul>
                        <li>2 days after click</li>
                        <li>or at maximum, the value of <code>attributionexpiry</code> + 1 hour after the click</li>
                    </ul>
                </td>
                <td>2</td>
            </tr>
            <tr>
                <td>Under 2 days</td>
                <td>
                    <ul>
                        <li>2 days + 1 hour after the click</li>
                    </ul>
                </td>
                <td>1</td>
            </tr>
        </tbody>
    </table>
</div>
```

Reviewer: I changed the wording of most rows here, look closer to verify it's alright

Author: Ty, I've used your table + made a small tweak (grey header).

 
### Attribution model
 
By default, the attribution model is **last-touch**: for a given conversion, only the last
engagement with an ad will generate a report.
 
You can change this behavior by configuring [source-side priorities](/docs/privacy-sandbox/attribution-reporting-event-guide/#define-custom-attribution-rules).
 
#### Example
 
A user visits `news.example` and clicks an ad for `shoes.example`. On the next day, they visit `other-news.example` and again click an ad for `shoes.example`. A few days later, they visit `shoes.example` and convert by making a purchase.

By default, unless you configure custom [source-side priorities](/docs/privacy-sandbox/attribution-reporting-event-guide/#define-custom-attribution-rules), you'll receive a report that attributes this conversion to the ad clicked on `other-news.example`.

### Matching source and trigger data

Let's take a given ad click (source event) with an click ID (source event ID) of `1298765678762`.
When this click is registered, the browser⏤on the user's device⏤stores the click ID `1298765678762`, and alongside it:

- `attributiondestination`. Typically, this is an advertiser site. Example: `shoes.example`.

> - `attributionreportto`. Typically, this is an adtech endpoint. Example: `adtech.example`.

Reviewer: Another good table candidate? With an extra column for the example

Author: SG, made it a table.

Later on, a trigger (conversion) takes place on a page.
The browser will match this trigger with the source event ID `1298765678762` in the following conditions:

> - The page on which the trigger (conversion) took place matches **the eTLD+1 that was provided in the source's `attributiondestination`**. Example: `shoes.example/checkout`.

Reviewer: Good place to link to https://web.dev/same-site-same-origin/#site ?

Author: Indeed! done

- The attribution is triggered (that is, the conversion is registered) via a request to the **same `attributionreportto` origin** (that then redirects to a `.well-known` location). Example: `adtech.example`.

<figure>

> {% Img src="image/O2RNUyVSLubjvENAT3e7JSdqSOx1/gdpjvYVB50vCbBUP2UKj.png", alt="ALT_TEXT_HERE", width="800", height="474" %}

Reviewer: Same alt comments as above

Author: Added!

 <figcaption>How the browser matches click IDs (source event IDs) with attribution trigger data (conversion data).</figcaption>
</figure>
 
Note:
 
- When a source is registered with a given (`attributiondestination`, `attributionreportto`) pair, existing sources matching this pair will be looked up in storage. If the matching sources have been triggered at least once (that is, have scheduled a report), they will be removed from browser storage and will not be eligible for further reporting. Any pending reports for these sources will still be sent.
- For click events, a source will be registered only if the resulting document being navigated to ends up sharing the eTLD+1 with the `attributiondestination` origin.

### User-initiated data clearing

Data clearing on the user's end⏤for example, when a user deletes their browsing history⏤can affect measurement based on the Attribution Reporting API in ways that differ from cookie-based measurement.

Head over to [FAQ: Impact of user-initiated data clearing on attribution reports](/docs/privacy-sandbox/attribution-reporting-data-clearing/) for details.

> ## Summary: Browser API parameters

Reviewer: Hm, not all of these are API parameters. Any ideas for a more appropriate subheading?

Author: Oh right, I went for 
`## Summary: API behavior and parameters`
LMK if this doesn't work!
